api/v1alpha2/postgresqluser_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// PostgreSQLUserSpec defines the desired state of PostgreSqlUser
+type PostgreSQLUserSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Server string `json:"server"`
+	// +kubebuilder:validation:Pattern=^[-\w\._\(\)]+$
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	ResourceGroup string `json:"resourceGroup"`
+
+	// optional
+
+	// The server-level roles assigned ot the user.
+	Roles []string `json:"roles,omitempty"`
+
+	// The databases of which this user should inherit the ownership role.
+	OwnedDatabases []string `json:"ownedDatabases,omitempty"`
```

Comment:
Hi @morelina - thanks for your work on this PR! 

Unfortunately, the way you're handling `ownedDatabases` here isn't compatible with the  way we were intending to do it in #1346, so if this lands as-is we'd eventually need to add another version to do it like that. (Unfortunately we've been focused on the code-generation work for v2 so we haven't gotten to this.)

Would you be willing to change it to work the way the `MySQLUser` type does? Rather than `OwnedDatabases []string`  there'd be `DatabaseRoles map[string][]string` of database name -> roles. You should be able to reuse the diffing code the MySQL reconciler uses to get the set of roles that have been added/removed, although the SQL to read and update those will be Postgres-specific.